### ISSUE TITLE
Lint entire project ignores .sqlfluffignore

### ISSUE THREAD

**Issue by Luttik** [sqlfluff/vscode-sqlfluff#72]
Hé @RobertOstermann 

The quick addition to this plugin where there is now a "lint entire project" action was great. There is one issue now however. It also tries to lint the files that are ignored via the .sqlfluffignore file.  This currently means that when I run this action in my DBT project I get an almost unending list of warnings that the <file that is ignored in .sqlfluff> is not a dbt model. In fact I get so many warnings that vscode just straight up crashes...

**Comment by RobertOstermann** [comment]
@Luttik Yeah there are a few issues with that setting. I will look into ignoring the files in the .sqlfluffignore file. I also will be improving the code to ensure it is less likely to crash in bigger projects. It may be a couple weeks before these changes are out.

**Comment by RobertOstermann** [comment]
@Luttik That was a great idea, had to make several changes to ensure the extension would lint workspaces correctly instead of linting single files, but running the command once is much more efficient and utilizes the sqlfluff ignore file correctly. Any time single files are linted that still does not respect the sqlfluff ignore file, might have to manually parse the file to fix that.

I just published a pre-release version of the extension that contains these linting improvements. Could you give that version a try and let me know if that fixes your problems?

**Comment by Luttik** [comment]
Hé Robert,

With version 2.2.0 installed, it does run nicely on the lint project action; it does run much faster and doesn't crash the IDE anymore 😉. 

The run on startup setting doesn't seem to work, though. There is no logging from sqlfluff in the vscode output. This indicates to me that there is not an error per sé but that the linting task is just not triggered (I might be wrong in that assessment though).

**Comment by RobertOstermann** [comment]
@Luttik I don't think there is a specific setting for run on startup. Is this happening with the `sqlfuff.linter.lintEntireProject = true`? The extension currently will not lint on startup if that is set to false, but if that is true it should lint the entire project on startup.
